src/psyclone/domain/lfric/kern_call_arg_list.py

```diff
@@ -379,11 +379,19 @@ def field(self, arg, var_accesses=None):
         # Look-up the name of the variable that stores the reference to
         # the data in this field.
         sym = self._symtab.lookup_with_tag(f"{arg.name}:{suffix}")
-        # Add the field data array as being read.
-        self.append(sym.name, var_accesses, var_access_name=sym.name,
-                    mode=arg.access, metadata_posn=arg.metadata_index)
 
-        self.psyir_append(Reference(sym))
+        if self._kern.iterates_over == "dof":
+            # If dof kernel, add access to the field by dof ref
```

Comment:
Hi @oakleybrunt, I think your problem is here. If you look at the failing example it's because it's failing to get any fields to upload to the GPU. I think this is because your modification here (which admittedly I've reviewed once) is only listing the dof loop symbol, not the field. This method should probably just revert to what it was originally.

Reply:
We should therefore have a new test for this method too.